be/src/exec/exec_node.cpp

```diff
@@ -795,6 +813,7 @@ void ExecNode::collect_scan_nodes(vector<ExecNode*>* nodes) {
     collect_nodes(TPlanNodeType::MYSQL_SCAN_NODE, nodes);
     collect_nodes(TPlanNodeType::LAKE_SCAN_NODE, nodes);
     collect_nodes(TPlanNodeType::SCHEMA_SCAN_NODE, nodes);
+    collect_nodes(TPlanNodeType::STREAM_SCAN_NODE, nodes);
```

Comment:
it's pretty inefficient to visit the plan tree several times. we should refactor it with a node set

Reply:
How about I refactor it in another PR?

Reply:
sure